From 4d0494574b2cda85ff0c6b968be362594c36aa37 Mon Sep 17 00:00:00 2001
From: Tom Hacohen <tom@stosb.com>
Date: Thu, 7 Apr 2016 14:56:57 +0100
Subject: [PATCH 26/79] Edje: Reparent when swallowing/unswallowing.

On swallow make the edje object the parent, on unswallow the canvas.
---
 src/lib/edje/edje_edit.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index 05ed301..5ddd4f4 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -315,6 +315,7 @@ _edje_real_part_free(Edje *ed, Edje_Real_Part *rp)
    if ((rp->type == EDJE_RP_TYPE_SWALLOW) && (rp->typedata.swallow)
        && (rp->typedata.swallow->swallowed_object))
      {
+        eo_parent_set(rp->typedata.swallow->swallowed_object, evas_common_evas_get(ed->obj));
         evas_object_smart_member_del(rp->typedata.swallow->swallowed_object);
         evas_object_event_callback_del(rp->typedata.swallow->swallowed_object,
                                        EVAS_CALLBACK_FREE, _edje_object_part_swallow_free_cb);
@@ -3147,7 +3148,11 @@ _edje_edit_real_part_add(Evas_Object *obj, const char *name, Edje_Part_Type type
         evas_object_show(rp->object);
         evas_object_smart_member_add(rp->object, ed->obj);
         evas_object_layer_set(rp->object, evas_object_layer_get(ed->obj));
-        if (ep->type != EDJE_PART_TYPE_SWALLOW && ep->type != EDJE_PART_TYPE_GROUP)
+        if (ep->type == EDJE_PART_TYPE_SWALLOW)
+          {
+             eo_parent_set(rp->object, ed->obj);
+          }
+        else if (ep->type != EDJE_PART_TYPE_GROUP)
           {
              if (ep->mouse_events)
                {
-- 
2.7.4

